commons/src/main/java/io/aiven/kafka/connect/common/grouper/TopicPartitionRecordGrouper.java

```diff
@@ -135,9 +139,20 @@ public void clear() {
         fileBuffers.clear();
     }
 
+    @Override
+    public void clearProcessedRecords(final String identifier, final List<SinkRecord> records) {
+        final GroupedSinkRecord grouperRecord = fileBuffers.getOrDefault(identifier, null);
```

Comment:
Are you trying to protect from keys with `null` values with `getOrDefault` or is there any other reason for this?

Reply:
I can't remember if I had any other specific reason for this other then readability, to be honest.